src/qt/peerdialog.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2011-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#if defined(HAVE_CONFIG_H)
+#include "config/bitcoin-config.h"
+#endif
+
+#include <iostream>
+#include <string>
+
+#include "peerdialog.h"
+
+#include "ui_addpeerdialog.h"
+#include "ui_removepeerdialog.h"
+#include "ui_testpeerdialog.h"
+
+#include "net.h"
+#include "net_processing.h"
+#include "netbase.h"
+#include "protocol.h"
+#include "ui_interface.h"
+#include "util.h"
+#include "guiconstants.h"
+#include "guiutil.h"
+
+#include <stdio.h>
+
+#include <QMessageBox>
+#include <QThread>
+
+/** Function to manage peers */
+QString PeerTools::ManagePeer(QString type, QString peer)
+{
+    std::string peerAddress = peer.toStdString();
+
+    if(!g_connman)
+        return "Error: Peer-to-peer functionality missing or disabled";
+
+    if (type == "onetry")
+    {
+        CAddress addr;
+        g_connman->OpenNetworkConnection(addr, false, NULL, peerAddress.c_str());
+        return "Attempted to one try node.";
+    }
+
+    if (type == "add")
+    {
+        if(!g_connman->AddNode(peerAddress))
+            return "Error: Node already added";
+    }
+    else if(type == "remove")
+    {
+        if(!g_connman->RemoveAddedNode(peerAddress))
+            return "Error: Node has not been added.";
+    }
+
+    return "Executed command.";
+}
+
+
+/** Add Peer Dialog */
+AddPeerDialog::AddPeerDialog(QWidget *parent) : 
+    QWidget(parent),
+    ui(new Ui::AddPeerDialog)
+{
+    ui->setupUi(this);
+
+    connect(ui->pushButton, SIGNAL(clicked()), this, SLOT(on_addPeer_clicked()));
+}
+
+AddPeerDialog::~AddPeerDialog()
+{
+    delete ui;
+}
+
+void AddPeerDialog::on_addPeer_clicked()
+{
+    QString peer = ui->lineEdit->text();
+    QString data = "";
+
+    if(peer.isEmpty()) 
+    {
+         QMessageBox::critical(this, "Add Peer", "Please enter an address.", QMessageBox::Ok, QMessageBox::Ok);
+         return;
+    }
+
+    data = peer.contains(":") ? peer : peer + ":" + "22556";
```

Comment:
2 issues with this:

1. an ipv6 peer has colons in the address part
2. standard port is 44556 for testnet

Perhaps it's user-friendlier to make 2 input fields - one for address, one for port - and remove the string interpretation altogether?

(same for line 120)

Reply:
You are right about the ipv6 address. I will add the 2 input fields in next commit.